nix/flake.nix

```diff
@@ -12,23 +12,31 @@
       };
     };
 
+    # NOTE: @2024-05 Nix flakes does not support getting git submodules of 'self'.
+    # refs:
+    # - https://discourse.nixos.org/t/get-nix-flake-to-include-git-submodule/30324
+    # - https://github.com/NixOS/nix/pull/7862
+    #
+    # ... In the meantime we kinda duplicate the dependencies here then replace the submodules with
+    # links to each repo in package sources.
+    #
+    # Try to use tags when possible to increase readability
+    # (note: `git submodule status` in wezterm repo will show the `git describe` result for each
+    # submodule, can help finding a tag if any)
     freetype2 = {
-      url = "github:wez/freetype2/de8b92dd7ec634e9e2b25ef534c54a3537555c11";
+      url = "github:wez/freetype2/e4586d960f339cf75e2e0b34aee30a0ed8353c0d";
```

Comment:
FWIW, you could use the tag for this too:

```
 e4586d960f339cf75e2e0b34aee30a0ed8353c0d deps/freetype/freetype2 (VER-2-13-1)
```

Reply:
hmm I don't see that tag 👀
https://github.com/wez/freetype2/tags